llvm/lib/Target/PowerPC/PPCISelLowering.cpp

```diff
@@ -3369,6 +3369,59 @@ SDValue PPCTargetLowering::LowerGlobalTLSAddressAIX(SDValue Op,
   bool Is64Bit = Subtarget.isPPC64();
   bool HasAIXSmallLocalExecTLS = Subtarget.hasAIXSmallLocalExecTLS();
   TLSModel::Model Model = getTargetMachine().getTLSModel(GV);
+  // Initialize TLS model opt setting lazily:
```

Comment:
This primarily sets the TLS model per function. Not sure if it is possible, but can it/does it make sense for this part to be separated into its own function?

Reply:
Sure. Created static function guarded by `Subtarget.hasAIXShLibTLSModelOpt()`. Thank you for the suggestion!

Reply:
Sure. Created static function guarded by `Subtarget.hasAIXShLibTLSModelOpt()`. Thank you for the suggestion!